packages/flutter_hooks/test/use_search_controller_test.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/src/framework.dart';
+import 'package:flutter_hooks/src/hooks.dart';
+
+import 'mock.dart';
+
+void main() {
+  testWidgets('debugFillProperties', (tester) async {
+    await tester.pumpWidget(
+      HookBuilder(builder: (context) {
+        useSearchController();
+        return const SizedBox();
+      }),
+    );
+
+    await tester.pump();
+
+    final element = tester.element(find.byType(HookBuilder));
+
+    expect(
+      element
+          .toDiagnosticsNode(style: DiagnosticsTreeStyle.offstage)
+          .toStringDeep(),
+      equalsIgnoringHashCodes(
+        'HookBuilder\n'
+        ' │ useSearchController:\n'
+        ' │   SearchController#00000(TextEditingValue(text: ┤├, selection:\n'
+        ' │   TextSelection.invalid, composing: TextRange(start: -1, end:\n'
+        ' │   -1)))\n'
+        ' └SizedBox(renderObject: RenderConstrainedBox#00000)\n',
+      ),
+    );
+  });
+
+  group('useSearchController', () {
+    testWidgets('initial values matches with real constructor', (tester) async {
+      late SearchController controller;
+      late SearchController controller2;
+
+      await tester.pumpWidget(
+        HookBuilder(builder: (context) {
+          controller = useSearchController();
+          controller2 = SearchController();
```

Comment:
What is the purpose of this?

Reply:
It probably could be created outside of the HookBuilder

Reply:
I tried to match the testcases of other hooks.